src/components/prompt/PromptBox.tsx

```diff
@@ -456,31 +483,43 @@ const PromptBox = forwardRef(function (
           </div>
         )}
       {/* ------ ellipses menu  ------- */}
-      {prompt.isSaved &&
-        (isHovered || isEditing) &&
-        !isContextCollapsed(context) &&
-        !isContextBulk(context) && (
-          <div
-            css={css`
-              position: absolute;
-              top: 3px;
-              right: 1px;
-            `}
-          >
-            <PromptEllipses onClick={() => setContextMenuOpen(true)} />
-          </div>
-        )}
+      {prompt.isSaved && (isHovered || isEditing) && !isContextBulk(context) && (
+        <div
+          css={css`
+            position: absolute;
+            top: ${isContextCollapsed(context) &&
+            collapsedDirection === CollapsedPromptDirection.RTL
+              ? "calc(100% - 32px)"
+              : "3px"};
+            right: ${isContextCollapsed(context) &&
+            collapsedDirection === CollapsedPromptDirection.RTL
+              ? 8
+              : 1}px;
+          `}
+        >
+          <PromptEllipses
+            onClick={() => {
+              determineContextLayout();
+              setContextMenuOpen(true);
+            }}
+          />
+        </div>
+      )}
       {/* ----- context menu ------ */}
       {contextMenuOpen && (
         <div
           css={css`
             position: absolute;
-            top: 0px;
-            left: calc(100%);
+            top: ${isContextCollapsed(context) &&
+            collapsedDirection === CollapsedPromptDirection.RTL
+              ? "calc(100% - 32px)"
+              : "0px"};
+            left: calc(100% - ${!contextMenuRightLayout ? "295px" : "0px"});
```

Comment:
we can play with the positioning here...

Reply:
Ah, sorry you got to this before I did! Will fine-tune…